operators/nftopology/controllers/packagerevision_controller.go

```diff
@@ -0,0 +1,258 @@
+		nfdeployedInst.NFType = string(nfTemplate.NFType)
+		nfdeployedInst.NFVendor = vendor
+		nfdeployedInst.NFVersion = version
+		nfdeployed.Spec.NFInstances = append(nfdeployed.Spec.NFInstances, nfdeployedInst)
+		nfdeployedIdx = len(nfdeployed.Spec.NFInstances) - 1
+	}
+
+	neighborMap := BuildNeighbormap(BuildAttachmentMap(nfTopo))
+	instMap := BuildDeployedInstanceMap(nfdeployed)
+
+	// this is the continuous update case; so first build the neighbor list for this instance, then
+	// for all the connected instance(s), update by appending the neighbor list
+	/*
+	 * TODO(s3wong): this now assumes each NFInstance from NFTopology.NFInstance will generate exactly
+	 * one NF instance, an assumption that should hold true for R1
+	 */
+	instName := GetNFInstanceNameFromNFDeployedName(nfDeployedName)
+	neighborSlice, _ := neighborMap[instName]
+	for _, neighbor := range neighborSlice {
+		if neighborIdx, ok := instMap[neighbor.Name]; !ok {
+			// neighbor packagerevision object not created yet
+			continue
+		} else {
+			neighborInst := &(nfdeployed.Spec.NFInstances[neighborIdx])
+			nfdeployedInstance := &(nfdeployed.Spec.NFInstances[nfdeployedIdx])
+			con := nfdeployv1alpha1.NFDeployedConnectivity{}
+			con.NeighborName = neighborInst.Id
+			nfdeployedInstance.Connectivities = append(nfdeployedInstance.Connectivities, con)
+			neighborCon := nfdeployv1alpha1.NFDeployedConnectivity{}
+			neighborCon.NeighborName = nfdeployedInstance.Id
+			neighborInst.Connectivities = append(neighborInst.Connectivities, neighborCon)
+		}
+	}
+	return nil
+}
+
+// +kubebuilder:rbac:groups=porch.kpt.dev,resources=packagerevisions,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=porch.kpt.dev,resources=packagerevisions/status,verbs=get;update;patch
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+func (r *PackageRevisionReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	l := log.FromContext(ctx).WithValues("NFTopology-PackageRevision", req.NamespacedName)
+
+	pkgRev := &porchv1alpha1.PackageRevision{}
+	if err := r.Get(ctx, req.NamespacedName, pkgRev); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// Step 1: check package revision state
+	// TODO(s3wong): if package deleted, inform ESA
+	if pkgRev.Spec.Lifecycle != porchv1alpha1.PackageRevisionLifecyclePublished {
+		l.Info("Package lifecycle is not Published, ignore", "PackageRevision.lifecycle",
+			pkgRev.Spec.Lifecycle)
+		return ctrl.Result{}, nil
+	}
+
+	var regionName, clusterName, nfInstanceName, topologyName, topoNamespace string
+	var createDeployed, found bool
+	pkgRevLabels := pkgRev.ObjectMeta.Labels
+	if regionName, found = pkgRevLabels["region"]; !found {
+		l.Info("region name not found from package")
+		return ctrl.Result{}, nil
+	}
+	if nfInstanceName, found = pkgRevLabels["nfinstance"]; !found {
```

Comment:
If these are ones you create, you should namespace them. Probably `nftopology.nephio.org/nfinstance`, etc.